esmvaltool/preprocessor/_area_pp.py

```diff
@@ -137,3 +137,42 @@ def area_average(cube, coord1, coord2):
     grid_areas = iris.analysis.cartography.area_weights(cube)
     result = cube.collapsed(coords, iris.analysis.MEAN, weights=grid_areas)
     return result
+
+
+def extract_named_regions(cube, regions):
+    """
+    Extract a specific named region.
+
+    The region coordinate exist in certain CMIP datasets.
+    This preprocessor allows a specific named regions to be extracted.
+
+    Arguments
+    ---------
+    cube: iris.cube.Cube
+       input cube.
+
+    regions: str, list
+        A region or list of regions to extract.
+
+    Returns
+    -------
+    iris.cube.Cube
+        collapsed cube.
+    """
+    # Make sure regions is a list of strings
+    if isinstance(regions, str):
+        regions = [regions, ]
+
+    if not isinstance(regions, list):
```

Comment:
There are other valid options than lists, e.g. tuple, set, etc.

Reply:
I'm assuming that there are only so many possible ways to write a list in yaml, then for ESMValTool to load that yml list into python. 

Replaced the line:
```
    if not isinstance(regions, list):
```
with:
```
    if not isinstance(regions, (list, tuple, set)):
```